addons/mail/static/src/utils.js

```diff
@@ -56,3 +56,37 @@ export function useAutoScroll(refName) {
         }
     });
 }
+
+export function useHover(refName) {
+    const ref = useRef(refName);
+    const state = useState({ isHover: false });
+    function onMouseMove(ev) {
+        if (ref.el && ref.el.contains(ev.target)) {
+            state.isHover = true;
+            return;
+        }
+        state.isHover = false;
+    }
+    useExternalListener(window, 'mousemove', onMouseMove, true);
+    return state;
+}
+
+export function useObserve(obj, key, callback) {
+    useEffect(
+        () => observe(obj, key, callback),
+        () => []
+    );
+}
+
+export function observe(obj, key, callback) {
+    let isActive = true;
+    const reactiveObj = reactive(obj, () => {
+        if (isActive) {
+            callback(reactiveObj[key]);
+        }
+    });
+    reactiveObj[key]; // initial subscription
```

Comment:
can prefix with `void` so eslint is happy (avoids `no-unused-expressions` warning)